qiskit/aqua/operators/operator_base.py

```diff
@@ -112,6 +113,16 @@ def to_matrix(self, massive: bool = False) -> np.ndarray:
         """
         raise NotImplementedError
 
+    @abstractmethod
```

Comment:
Currently this is defined for PrimitiveOp - this moves it up the heirarchy to include ListOp based forms. Given my comment to ListOp above I am not so sure on this. I would appear that ListOp itself possibly does not meaningfully support this -  sure perhaps one could concatenate all the circuits in the list into one large one but to what end.